### ISSUE TITLE
Is nested-interactive still a relevant rule?

### ISSUE THREAD

**Issue by gmascia** [dequelabs/axe-core#3950]
### Product

axe-core

### Question

The assumption that justifies the existence of the nested-interactive violation does not seem to be true anymore. 

The very example snippet provided in the rule's [page](https://dequeuniversity.com/rules/axe/4.6/nested-interactive?application=RuleDescription) does not get read out in the assumed incomplete/confusing way. The 'empty tab stop' issue does not seem to be reproducible. Nested interactive elements are announced exactly the same way as they would when not nested (role and status included). 

Try for example: https://jsfiddle.net/tzf5uagh/3/ (verified on VoiceOver/Chrome/Mac, Jaws/Chrome/Win, NVDA/Chrome/Win).

**Comment by straker** [comment]
Thanks for the question. It's still very much needed, especially when you try to nest ARIA role elements inside a intractable HTML elements. For example

```html
<button>
  Hello
  <div role="button" tabindex="0">Nested Button</div>
</button>
```

This will cause an empty tab stop in VO/Safari and NVDA/Firefox, and cause the button to read twice in JAWS/Chrome. 

P.S. Using ARIA roles and interactive HTML elements is the only case this rule applies to as nesting two interactive HTML elements is invalid HTML.
